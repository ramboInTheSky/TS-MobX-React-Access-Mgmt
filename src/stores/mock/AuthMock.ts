export class AuthMock {
    public user: any = {
        accessToken: 'xZXnpwklGhqd-zN5pJKbkHXNg5jBdJYu',
        idToken:
            'eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiIsImtpZCI6Ik5UY3â€¦XhKpVLcBSw6qUe4KgjCI4sIgDa3_Kh7TbaSLgd__pJdZLriQg',
        idTokenPayload: {},
        appState: null,
        refreshToken: null,
    }
    public login = jest.fn()
    public checkSession = jest.fn()
    public logout = jest.fn()
    public getUser = jest.fn(() => this.user)
    public redirectToPage = '/profile/123'
}
