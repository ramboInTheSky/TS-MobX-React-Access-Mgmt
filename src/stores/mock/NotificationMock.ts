export class NotificationMock {
    public setNotification: Function = jest.fn()
}
