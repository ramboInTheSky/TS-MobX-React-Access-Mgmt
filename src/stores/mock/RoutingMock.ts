import { Routing } from '../Routing'

export class RoutingMock extends Routing {
    public goToPage: any = jest.fn()
    public goBackToPage: any = jest.fn()
    public push: any = jest.fn()
    public replace: any = jest.fn()
    public go: any = jest.fn()
    public goBack: any = jest.fn()
    public goForward: any = jest.fn()
    public breadcrumb: any[] = [{ page: 'profile', id: '123-123-123' }]
    public location: any = {}
    public history: any = {}
    public getBreadcrumb: any = jest.fn(() => this.breadcrumb)
    public resizeBreadcrumb: any = jest.fn()
    public resizeBreadcrumbs: any = jest.fn()
    public splitThings: any = jest.fn()
    public rebuildBreadcrumb: any = jest.fn()
}
