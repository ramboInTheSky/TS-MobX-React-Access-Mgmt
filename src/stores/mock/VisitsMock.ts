import { Archetype, Visit } from '../../models/Visit'
import { Tag } from '../../models/Tag'

export class VisitsMock {
    public archetypes: {
        primary: Archetype[]
        secondary: Archetype[]
    } = {
        primary: [
            {
                id: '111-111-111',
                lockGroupNames: ['door1', 'door2'],
                name: 'Functional API Tests',
                primary: false,
            },
        ],
        secondary: [],
    }
    public getArchetypes: Function = jest.fn()
    public item: Visit = new Visit()
    public detail: any = {
        id: 'd8be7c8d-a89f-4632-b71e-2d756463643c',
        from_date: '2019-05-30T13:44:43.463Z',
        to_date: null,
        tag_numbers: ['00123123'],
        archetype_ids: ['d8be7c8d-a89f-4632-b71e-2d756463643c'],
    }
    public getDetail: Function = jest.fn(() => this.detail)
    public poll: Function = jest.fn(() => null)
    public addTag = jest.fn((tag: Tag, profileId: string, visitId: string) => {
        this.detail.tagNumbers.push(tag.tagNumber!)
    })
    public removeTag = jest.fn()
    public clearErrors = jest.fn()
    public addAccess = jest.fn()
    public removeAccess = jest.fn()
    public save = jest.fn()
    public validationError?: string = undefined
    public toggleEditing = jest.fn()
    public simulateValidationError = () =>
        (this.validationError = 'this is an error')
}
