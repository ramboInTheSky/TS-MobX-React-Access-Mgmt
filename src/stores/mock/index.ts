export * from './RoutingMock'
export * from './ProfilesMock'
export * from './VisitsMock'
export * from './AuthMock'
export * from './NotificationMock'

export const mockDateConstructor = () => {
    const now = new Date('2020')
    /*eslint no-global-assign:off*/
    Date = class extends Date {
        constructor() {
            super()
            return now
        }
    } as DateConstructor
}
