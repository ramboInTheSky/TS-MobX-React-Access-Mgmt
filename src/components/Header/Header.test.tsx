import React from 'react'
import { shallow } from 'enzyme'

import { Header } from '.'
import { AuthMock } from '../../stores/mock/AuthMock'

jest.mock('../../../package.json', () => ({
    version: 'this is a version number',
}))

describe('<Header />', () => {
    let props: any
    beforeEach(() => {
        props = {
            auth: new AuthMock(),
        }
    })
    it('renders correctly', () => {
        expect(shallow(<Header {...props} />)).toMatchSnapshot()
    })

    it('Does open the settings menu', () => {
        const el: any = shallow(<Header {...props} />)
        el.setState({ settingsMenu: true }, () => el.update())
        expect(el).toMatchSnapshot()
    })

    it('Does open the user menu', () => {
        const el: any = shallow(<Header {...props} />)
        el.setState({ userMenu: true }, () => el.update())
        expect(el).toMatchSnapshot()
    })

    it('Does logout', () => {
        const el: any = shallow(<Header {...props} />)
        el.setState({ userMenu: true }, () => el.update())
        const logoutItem = el.dive().find('#logout-menu-item')
        logoutItem.simulate('click', {
            preventDefault: () => {},
            target: { name: 'input', value: 'value' },
        })
        expect(props.auth.logout).toHaveBeenCalled()
    })
})
