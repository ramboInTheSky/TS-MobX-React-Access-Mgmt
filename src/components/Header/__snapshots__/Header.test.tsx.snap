// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> Does open the settings menu 1`] = `
<Header
  auth={
    AuthMock {
      "checkSession": [MockFunction],
      "getUser": [MockFunction],
      "login": [MockFunction],
      "logout": [MockFunction],
      "redirectToPage": "/profile/123",
      "user": Object {
        "accessToken": "xZXnpwklGhqd-zN5pJKbkHXNg5jBdJYu",
        "appState": null,
        "idToken": "eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiIsImtpZCI6Ik5UY3…XhKpVLcBSw6qUe4KgjCI4sIgDa3_Kh7TbaSLgd__pJdZLriQg",
        "idTokenPayload": Object {},
        "refreshToken": null,
      },
    }
  }
/>
`;

exports[`<Header /> Does open the user menu 1`] = `
<Header
  auth={
    AuthMock {
      "checkSession": [MockFunction],
      "getUser": [MockFunction],
      "login": [MockFunction],
      "logout": [MockFunction],
      "redirectToPage": "/profile/123",
      "user": Object {
        "accessToken": "xZXnpwklGhqd-zN5pJKbkHXNg5jBdJYu",
        "appState": null,
        "idToken": "eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiIsImtpZCI6Ik5UY3…XhKpVLcBSw6qUe4KgjCI4sIgDa3_Kh7TbaSLgd__pJdZLriQg",
        "idTokenPayload": Object {},
        "refreshToken": null,
      },
    }
  }
/>
`;

exports[`<Header /> renders correctly 1`] = `
<Header
  auth={
    AuthMock {
      "checkSession": [MockFunction],
      "getUser": [MockFunction],
      "login": [MockFunction],
      "logout": [MockFunction],
      "redirectToPage": "/profile/123",
      "user": Object {
        "accessToken": "xZXnpwklGhqd-zN5pJKbkHXNg5jBdJYu",
        "appState": null,
        "idToken": "eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiIsImtpZCI6Ik5UY3…XhKpVLcBSw6qUe4KgjCI4sIgDa3_Kh7TbaSLgd__pJdZLriQg",
        "idTokenPayload": Object {},
        "refreshToken": null,
      },
    }
  }
/>
`;
